### PR TITLE
[info arch] Add "search" filter to ExampleIndex

### PR DESCRIPTION
**Pull request by katydecorah**
This PR introduces a search to the filters in ExampleIndex that will allow the user to only display cards that have their text query in the title or description of the collection of cards.

> 2020-11-18 at 4 24 PM

## How to test

- Try out /PageLayout test case: `Example (index) layout.` 
- Try searching for different words in the title and the description of the cards, try lowercase and uppercase (search should be case insensitive).
- Do you have better naming suggestions for the label and placeholder? I'm not sure if I love it right now, but couldn't think of other ideas 🤔 

## QA checklist

- [x] No errors logged to console.
- [x] Accessibility score in [Chrome DevTools Audit/Lighthouse](https://developers.google.com/web/tools/lighthouse#devtools) is 100% with Lighthouse version: `#.#.#`.
- [x] Component is accessible at mobile-size viewport.
- [x] Component is accessible at tablet-size viewport.
- [x] Component is accessible at laptop-size viewport.
- [x] Component is accessible at big-monitor-size viewport.
- [ ] Create a PR in a site repo, copy the component, and test it. Push to staging and let the reviewer know they can also test the component there.

Open the test cases app locally on:

- [x] Chrome, no errors logged to console.
- [x] Firefox, no errors logged to console.
- [x] Safari, no errors logged to console.
- [x] Edge, no errors logged to console.
- [x] IE11, no errors logged to console.
- [x] Mobile Safari, no errors logged to console.
- [x] Android Chrome, no errors logged to console.

### CODE REVIEW

File: src/components/page-layout/components/example-index.js [hunk 1]
```diff
@@ -158,6 +166,16 @@ export default class ExampleIndex extends React.PureComponent {
...
         <AsideHeading>Filters</AsideHeading>
         <div className="grid grid--gut6">
+          {availableFilters.search && (
+            <FilterSection
+              title="Search"
+              id="search"
+              activeItem={search}
+              handleInput={this.handleInput}
+              isText={true}
+              placeholder="Search title and description&hellip;"
```
Comment (danswick) [inline src/components/page-layout/components/example-index.js:176]: Maybe "Search for an example..."?

**Review by danswick — APPROVED** [review]
This looks great! Tested locally and everything works as expected. I left a suggestion for placeholder for you to consider!

File: src/components/page-layout/components/example-index.js (hunk 1 above)
Comment (katydecorah) [reply src/components/page-layout/components/example-index.js:176]: Not sure we want to use "example" as filters can show up on tutorials and troubleshooting pages too 🤔

Comment (katydecorah) [reply src/components/page-layout/components/example-index.js:176]: Let's leave for now and update later if we find an alt!

**Pull request merged**